### PR TITLE
gadget/install/install_test: add unit tests for MountVolumes

### PR DESCRIPTION
**Pull request by ZeyadYasser**
There was no unit tests for [MountVolumes](https://github.com/snapcore/snapd/blob/85cdda5c732c6e6de04f522a5b076badb0c7b393/gadget/install/install.go#L566) before.

### CODE REVIEW

File: gadget/install/install_test.go [hunk 1]
```diff
@@ -1165,3 +1166,215 @@ func (s *installSuite) TestInstallEncryptPartitionsNoDeviceSet(c *C) {
...
+	_, _, err := install.MountVolumes(onVolumes, nil)
+	c.Assert(err, ErrorMatches, "there are 2 system-seed{,-null} partitions, expected one")
+
+	c.Assert(mountCall, Equals, 2)
+	// check unmount is called implicitly on error for cleanup
+	c.Assert(umountCall, Equals, 2)
+}
+
+func (s *installSuite) TestMountVolumesLazyUnmount(c *C) {
+	seedMntPt := filepath.Join(s.dir, "run/mnt/ubuntu-seed")
```
Comment (mvo5) [inline gadget/install/install_test.go:1340]: Bonus points if you could do additionally:
```go
log, restore := logger.MockLogger()
defer restore()
...
c.Check(log.String(), Contains, "cannot umount ....: forcing lazy umount (trying lazy umount next")
```

File: gadget/install/install_test.go [hunk 2]
```diff
@@ -1165,3 +1166,215 @@ func (s *installSuite) TestInstallEncryptPartitionsNoDeviceSet(c *C) {
...
+			return fmt.Errorf("forcing lazy unmount")
+		} else {
+			// check fallback to lazy unmount, see LP:2025402
+			c.Assert(flags, Equals, syscall.MNT_DETACH)
+			return nil
+		}
+	})
+	defer restore()
+
+	seedMntDir, unmount, err := install.MountVolumes(onVolumes, nil)
```
Comment (mvo5) [inline gadget/install/install_test.go:1371]: Maybe one extra test that checks that the correct error is returned if both umount() calls fail?

**Review by mvo5 — APPROVED** [review]
Thanks, this is very nice! Two small suggestions inline

File: gadget/install/install_test.go (hunk 1 above)
Comment (ZeyadYasser) [reply gadget/install/install_test.go:1341]: Updated, Thanks!

Comment (ZeyadYasser) [reply gadget/install/install_test.go:1341]: a side note, I noticed that the [fallback logic](https://github.com/snapcore/snapd/blob/85cdda5c732c6e6de04f522a5b076badb0c7b393/gadget/install/install.go#L576) will always log a problem even if fallback works. It's out of the scope of this PR, but I thought I mention it. It should already be fixed in this [PR](https://github.com/snapcore/snapd/pull/12967/files).

File: gadget/install/install_test.go (hunk 2 above)
Comment (ZeyadYasser) [reply gadget/install/install_test.go:1375]: Added, Thank you!

**Review by MiguelPires — APPROVED** [review]

**Pull request merged**